### ISSUE TITLE
Decimal point is recognized as 8

### ISSUE THREAD

**Issue by minyukim** [auerswal/ssocr#26]
Hello, my `1.111` is recognized as `18111` with below command
`ssocr -d -1 -D -v -P -W 1 -H 3 -f white out.png`

![out](https://github.com/auerswal/ssocr/assets/105956521/9190723f-c9ae-4ed0-8de9-bad6bed6ee14)

I thought I should be able to handle this case with -W and -H option, but failed because the width of `.` is almost same with `1`.
Is there any way to recognize this image correctly?

Below is the debug info
```
================================================================================
flags & VERBOSE=4
thresh=50.000000
flags & PRINT_INFO=0
flags & ADJUST_GRAY=0
flags & ABSOLUTE_THRESHOLD=0
flags & DO_ITERATIVE_THRESHOLD=0
flags & USE_DEBUG_IMAGE=8
flags & DEBUG_OUTPUT=0
flags & PROCESS_ONLY=0
flags & ASCII_ART_SEGMENTS=0
flags & PRINT_AS_HEX=0
flags & OMIT_DECIMAL=0
flags & PRINT_SPACES=0
flags & SPC_USE_AVG_DST=0
need_pixels = 1
min_segment = 1
min_char_dims = 1x1
ignore_pixels = 0
expected_digits.min = -1
expected_digits.max = -1
foreground = 255 (white)
background = 0 (black)
luminance  = Rec709
charset    = full
height/width threshold for one    = 3
width/height threshold for minus  = 2
max_dig_h/h threshold for decimal = 3
max_dig_w/w threshold for decimal = 1
distance factor for adding spaces = 1.40
optind=12 argc=13
================================================================================
argv[argc-1]=out.png used as image file name
loading image out.png
image width: 412
image height: 78
0.00 <= lum <= 255.00 (lum should be in [0,255])
adjusting threshold to image: 50.000000 -> 50.000000
using threshold 50.00
no commands given, using image out.png unmodified
starting image segmentation
starting horizontal partitioning
 start of potential digit 0 in image column 142
 end of potential digit 0 in image column 150
 start of potential digit 1 in image column 161
 end of potential digit 1 in image column 168
 start of potential digit 2 in image column 204
 end of potential digit 2 in image column 212
 start of potential digit 3 in image column 251
 end of potential digit 3 in image column 259
 start of potential digit 4 in image column 298
 end of potential digit 4 in image column 306
horizontal partitioning found 5 digit(s)
looking for upper and lower digit boundaries
image segmentation found 5 potential digits
image segmentation found 5 digits
total width of digit area is 164
total height of digit area is 48
digits are at most 8 pixels wide and 48 pixels high
found 5 digits
digit 0: (142,18) -> (150,65), width: 8 ( 4.88%) height: 47 (97.92%)
  height/width (int): 5, max_dig_w/width (int): 1, max_dig_h/height (int): 1
digit 1: (161,59) -> (168,65), width: 7 ( 4.27%) height: 6 (12.50%)
  height/width (int): 0, max_dig_w/width (int): 1, max_dig_h/height (int): 8
digit 2: (204,18) -> (212,65), width: 8 ( 4.88%) height: 47 (97.92%)
  height/width (int): 5, max_dig_w/width (int): 1, max_dig_h/height (int): 1
digit 3: (251,18) -> (259,66), width: 8 ( 4.88%) height: 48 (100.00%)
  height/width (int): 6, max_dig_w/width (int): 1, max_dig_h/height (int): 1
digit 4: (298,19) -> (306,66), width: 8 ( 4.88%) height: 47 (97.92%)
  height/width (int): 5, max_dig_w/width (int): 1, max_dig_h/height (int): 1
looking for digit 1
 digit 0 is a 1 (height/width = 47/8 = (int) 5)
 digit 2 is a 1 (height/width = 47/8 = (int) 5)
 digit 3 is a 1 (height/width = 48/8 = (int) 6)
 digit 4 is a 1 (height/width = 47/8 = (int) 5)
looking for decimal points
looking for minus signs
starting scanline based recognition for remaining digits
Display as seen by ssocr:
      _             
   | |_|   |   |   |
   | |_|   |   |   |

18111
using png format for debug image
writing debug image to file testbild.png
```

**Comment by auerswal** [comment]
Thanks for the report, I think this is a bug, i.e., currently `ssocr` has no sensible settings to recognize this image (the command `rotate 8` helps with *this* image, but I do not suggest it as a general workaround). I'll look into this.

**Comment by auerswal** [comment]
I have just released version 2.24.0 of `ssocr` that should fix this issue. The fix is the introduction of a special case for decimal point recognition when the widest digit in the image has been recognized as a one.

I had also considered to allow `--dec-w-ratio=0`, which would have worked with your image, too, but using this would mean recognizing minus signs as decimal points. Thus `--dec-w-ratio=0` is still ignored.

Please let me know if this works for you.

**Comment by minyukim** [comment]
It works for me!
Thank you so much for the quick fix and detailed explanation 👍
